designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/CustomActionValidator.scala

```diff
@@ -0,0 +1,129 @@
+package pl.touk.nussknacker.ui.validation
+
+import cats.data.{Validated, ValidatedNel}
+import cats.implicits._
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.context.PartSubGraphCompilationError
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.MismatchParameter
+import pl.touk.nussknacker.engine.api.definition.MandatoryParameterValidator
+import pl.touk.nussknacker.engine.api.deployment.CustomActionCommand
+import pl.touk.nussknacker.engine.api.parameter.ParameterName
+import pl.touk.nussknacker.engine.deployment.{CustomActionDefinition, CustomActionParameter}
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.restmodel.CustomActionRequest
+
+class CustomActionValidator(allowedAction: CustomActionDefinition) {
```

Comment:
Do you see an option to unify this with `ScenarioPropertiesValidator`? It is a huge duplication of quite easy to broken business logic. Also users may be disgusted if they find out that this logic behave different for some corner cases.

Reply:
I think in the current state we could possibly merge these two in a separate refactor. I don't think that's a good idea, because ScenarioProperties is a different object from CustomAction. I think merging this makes the code unnecesarily complex, especially if it turns out that the logic isn't exactly the same or we have to change it in the future. It's just 90 extra lines. 
Your user argument is convincing, but we should just make sure the external results are the same.